homeassistant/components/knx/config_flow.py

```diff
@@ -193,6 +219,83 @@ async def async_step_manual_tunnel(
             step_id="manual_tunnel", data_schema=vol.Schema(fields), errors=errors
         )
 
+    async def async_step_secure_manual(
+        self, user_input: dict | None = None
+    ) -> FlowResult:
+        """Configure ip secure manually."""
+        errors: dict = {}
+
+        if user_input is not None:
+            entry_data: KNXConfigEntryData = {
+                **self._tunneling_config,  # type: ignore[misc]
+                CONF_KNX_SECURE_USER_ID: user_input[CONF_KNX_SECURE_USER_ID],
+                CONF_KNX_SECURE_USER_PASSWORD: user_input[
+                    CONF_KNX_SECURE_USER_PASSWORD
+                ],
+                CONF_KNX_SECURE_DEVICE_AUTHENTICATION: user_input[
+                    CONF_KNX_SECURE_DEVICE_AUTHENTICATION
+                ],
+                CONF_KNX_CONNECTION_TYPE: CONF_KNX_TUNNELING_TCP_SECURE,
+            }
+
+            return self.async_create_entry(
+                title=f"Secure {CONF_KNX_TUNNELING.capitalize()} @ {self._tunneling_config[CONF_HOST]}",
```

Comment:
I think you need to add `assert self._tunneling_config` to satisfy mypy here

Reply:
Thanks, fixed both, not sure why my local pre-commit didnt find this though